dts/bindings/base/zephyr,memory-region.yaml

```diff
@@ -24,6 +24,8 @@ properties:
       - "RAM"
       - "RAM_NOCACHE"
       - "FLASH"
+      - "PPB"
+      - "IO"
```

Comment:
While the previous values were pretty self explanatory, these new ones might require some details. Or maybe a pointer to the whole classification

Reply:
@GeorgeCGV can you address this comment?

Reply:
@carlocaione, I am not sure how. Unfortunately, the opriginal author wasn't a very chatty person based on the: https://github.com/zephyrproject-rtos/zephyr/blob/c534ae294724f3463369363da9818a7a3cf81f95/include/zephyr/arch/arm/aarch32/mpu/arm_mpu_v7m.h#L137-L139

IO is quite self-explanatory and PPB is probably a Private Peripheral Bus. I am not very deep into this topic myself. However, I assume that PPB and IO memory region assumes certain rights and sharing settings.

Ideas?

Reply:
This raises more questions really. Are you using this code or not?

Reply:
Ok gotcha. Then just add a comment in the yaml file and in the header file describing that the values are taken from `arm_mpu_v7m.h`

Reply:
> comment in the yaml file

Isn't it already documented and implied by `devicetree_regions.h`?

https://github.com/zephyrproject-rtos/zephyr/blob/1839c275264b3919db669b77d8c90af949d4246d/include/zephyr/linker/devicetree_regions.h#L236-L243

>  in the header file

?

Reply:
Done

Reply:
I still think we miss a comment in this binding to link with the code (It's not obvious that these defines hook to `include/zephyr/linker/devicetree_regions.h`)
This being said, I agree this is not specific to this PR.